### DISCUSSION TITLE
[@types/openui5] Method can return `undefined` but type does not contain it: `Controller.byId` & `Router.getRoute`.

### DISCUSSION THREAD

**Discussion by sz3lbi** [DefinitelyTyped/DefinitelyTyped#61448] [Issues with a @types package] [answered]
Hello,
According to the documentation provided by IDE, the `Router.getRoute` method:
>Returns the route with the given name or undefined if no route is found.

>@returns — Route with the provided name or undefined.

Nevertheless, the returning type is `Route`, not `Route | undefined`

The same issue occurs with the `Controller.byId` method:
> If no view is connected or if the view doesn't contain an element with the given local ID, undefined is returned.

But `UI5Element`, not `UI5Element | undefined` is returned.

The issue is present in the OpenUI5 types, as well as the SAPUI5 types.
Thank you in advance for the fix.

[a comment by an automated account is not shown]

**Reply by codeworrior** [reply]
These are caused by documentation weaknesses in the OpenUI5 source code. We'll look into this.

**Comment by akudev** [comment, marked as answer]
@codeworrior has fixed these and a few more yesterday in https://github.com/SAP/openui5/commit/78e6c31c36068a62de7df591b8d2c10d925445c7 (will be part of the upcoming 1.105 release).

Several dozen more such fixes (~80) are in the pipeline, but might only make it into 1.106.

Thanks for reporting, @sz3lbi !

You are welcome to report issues with the UI5 type definitions at https://github.com/SAP/ui5-typescript/issues or - if they stem from the JSDoc in the original OpenUI5 source code, from which the type definitions are generated - directly at https://github.com/SAP/openui5/issues
